### DISCUSSION TITLE
Saving an Image/Content other than Text using Tauri

### DISCUSSION THREAD

**Discussion by louremipsum** [tauri-apps/tauri#6535] [Q&A] [answered]
I want to save an image which will come from a URL and I read the docs and understood that i would have to use (writeBinaryFile)[https://tauri.app/v1/api/js/fs/#writebinaryfile] to save an image to the system but I don't understand how i should convert the image which will come from the web to `BinaryFileContent` or do i even have to do that too?

this is my example code for saving
![image](https://user-images.githubusercontent.com/72456774/227099900-9507fef7-9a0b-4cc0-99c7-261c7366d87d.png)

I'm confused how to use `writeBinaryFile` and where should I put the image content args due to only 1 example given in docs

**Comment by FabianLars** [comment, marked as answer]
Doesn't your code work? It looks correct to me at first glance.

Anyway the _upload_ plugin also has a _download_ function: <https://github.com/tauri-apps/plugins-workspace/tree/dev/plugins/upload> which i'd recommend in use-cases like this because it doesn't pipe the data through the IPC so it should be faster.

**Reply by louremipsum** [reply]
my implementation saves a file of 0 bytes which doesn't open 
I did see the upload plugin and also saw your [comment](https://github.com/tauri-apps/tauri/discussions/4726#discussioncomment-3196765) on one other discussion where you said that there would be a download option soon but because there is nowhere written in the docs or the name of the upload plugin I thought it wasn't released yet.

Also, the plugin works perfectly, Thank you 😄 
ps. I quite like working using Tauri and after my current project, I want to contribute to Tauri Docs very much to make them beginner-friendly.

**Reply by FabianLars** [reply]
we love every kind of contributions and especially documentation ones are much much appreciated! ❤️

**Reply by louremipsum** [reply]
I wanted to make a download manager but I feel like this plugin may not be enough for example the ability to pause a download or concurrently download more than 1 file etc ... 
is there any package that you would recommend that would fit well with Tauri? Rust or JS package whichever would be better

**Reply by FabianLars** [reply]
I think that would require manual work in rust, i'm not aware of any existing crate that can pause downloads. And JS packages are a bit out of the questions because of the performance problems (and i'd imagine most of the packages that could do this need a nodejs runtime and can't work in a browser).
